crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1416,10 +1416,20 @@ impl<'db> ClassType<'db> {
         let DefinitionKind::Class(class_stmt_node) = definition.kind(db) else {
             panic!("Class type definition must have DefinitionKind::Class");
         };
+        let model: SemanticModel<'db> = SemanticModel::new(db, definition.file(db));
```

Comment:
We could create the `SemanticModel` only within the branch below where we'll actually need it.